discovery-infra/tests/test_day2.py

```diff
@@ -0,0 +1,40 @@
+import day2
+import pytest
+from junit_report import JunitTestSuite
+from tests.base_test import BaseTest
+from tests.config import ClusterConfig, global_variables
+from types import SimpleNamespace
+
+
+class TestDay2(BaseTest):
+    @pytest.fixture
+    def new_cluster_configuration(self):
+        return self.override_cluster_configuration()
+
+    def override_cluster_configuration(self):
+        config = ClusterConfig()
+        config.cluster_id = global_variables.cluster_id
+        return config
+
+    # Install day1 cluster and deploy day2 nodes (cloud flow).
+    # Or, deploy day2 nodes on an installed cluster if CLUSTER_ID env var is specified.
+    @JunitTestSuite()
+    def test_deploy_nodes_cloud(self, cluster):
+        if not global_variables.cluster_id:
+            cluster.prepare_for_installation()
+            cluster.start_install_and_wait_for_installed()
+
+        args = SimpleNamespace()
+        args.api_client = cluster.api_client
+        args.pull_secret = cluster.config.pull_secret
+        args.with_static_network_config = cluster.config.is_static_ip
+        args.num_day2_workers = global_variables.num_day2_workers
+        args.ssh_key = global_variables.ssh_public_key
+        args.install_cluster = True
+
+        if cluster.config.proxy:
+            args.http_proxy = cluster.config.proxy.http_proxy
+            args.https_proxy = cluster.config.proxy.https_proxy
+            args.no_proxy = cluster.config.proxy.no_proxy
+
+        day2.execute_day2_cloud_flow(cluster.config.cluster_id, args, cluster.config.is_ipv6)
```

Comment:
`day2.py` is deprecated. Add this logic to another module or create a new one. Once this task is done we want to delete `day2.py` module
Maybe create a new class for `Day2Cluster` that inherits from `Cluster` and implement the extra logic

Reply:
Also I think that this function signature should get cluster and cluster config

Reply:
Regarding the day2.py deprecation, added a note in the description that for used day2.py mostly as-is. Since it requires a rewrite of the logic and integration into tests, it's safer to handle it after we have a running e2e test.